app/assets/javascripts/discourse/app/widgets/post-menu.js

```diff
@@ -377,7 +377,7 @@ registerButton("delete", (attrs) => {
       icon: "far-trash-alt",
       className: "delete",
     };
-  } else if (attrs.showFlagDelete) {
+  } else if (attrs.showFlagDelete && attrs.canFlag) {
```

Comment:
Maybe the `canFlag` check can be in `lib/transform-post.js`, where `showFlagDelete` is set?

Reply:
Good catch!